src/AuthenticationController.js

```diff
@@ -148,7 +148,7 @@ class Authentication extends AbstractController {
           next()
         })
       })
-      .catch(error => next(error))
+      .catch(error => res.status(400).json({ error }))
```

Comment:
Wasn't this supposed to be a 409?

Also `error` is a javascript error which we don't want to send back in the response. This should rather be something like `{ error: error.message }` to produce:
```
{
  "error": "User already exists"
}
```

Reply:
It will catch both errors so 409 will only make sense when it catches a username conflict but if it catches `Registration Failed` it would still respond with a 409 even though that might not be the case? It seemed better to make it a more general 400 due to that.

The .json({ error }) is just the `.toString()` of the error object. So that in development it contains the stack trace but in production it is just the error message.

Reply:
Fair enough. Looks good to me in that case.